### PR TITLE
[extractor/gmanetwork] Add extractor

### LINKED ISSUE #5770: GMA Network
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I'm running yt-dlp version **2022.11.11** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

Worldwide

### Example URLs

https://www.gmanetwork.com/

https://www.gmanetwork.com/fullepisodes/

https://www.gmanetwork.com/fullepisodes/home/more_than_words/87059/more-than-words-full-episode-80/video?section=home

https://www.gmanetwork.com/fullepisodes/home/running_man_philippines/168677/running-man-philippines-catch-the-thief-full-chapter-2/video?section=home

https://www.gmanetwork.com/music/artists/gmamusic/165/sanyalopez/artist/ => https://aphrodite.gmanetwork.com/records/songs/Hot-Maria-Clara.mp3

https://www.gmanetwork.com/radio/streaming/dzbb

https://www.gmanetwork.com/regionaltv/

https://www.gmanetwork.com/entertainment/tv/kapuso_stream/videos/#176114

### Provide a description that is worded well enough to be understood

The site is a Philippines based TV network, and which is currently not supported by the yt-dlp.

I have seen some videos are embedded from YouTube or daily motion.

Please have a look into the site for the possible support.

There are few exclusive contents like tv dramas  restricted to south east asia or around Philippines due to the license (eg. https://www.gmanetwork.com/gtv/drama/the_merciless_judge/175785/the-merciless-judge-the-devil-judge-becomes-an-evil-doctor-episode-9/video).

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
yt-dlp -vU https://www.gmanetwork.com/fullepisodes/home/running_man_philippines/168677/running-man-philippines-catch-the-thief-full-chapter-2/video?section=home
[debug] Command-line config: ['-vU', 'https://www.gmanetwork.com/fullepisodes/home/running_man_philippines/168677/running-man-philippines-catch-the-thief-full-chapter-2/video?section=home']
[debug] User config "C:\Users\USER\AppData\Roaming\yt-dlp\config.txt": ['-o', 'E:\\yt-dlp_dls\\%(title)s.%(ext)s', '--sub-langs', 'en.*', '--extractor-args', 'youtube:skip=translated_subs', '--embed-subs', '-f', '((137/136)+140)/mpd-4/1/500/(hls-3462/hls-3192/hls-1960/hls-1660)', '--merge-output-format', 'mp4', '--ignore-no-formats-error', '-i', '--no-abort-on-error']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.11.11 [8b64402] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19043-SP0 (OpenSSL 1.1.1k  25 Mar 2021)

[debug] exe versions: ffmpeg n5.1.2-9-g807afa59cc-20221210 (setts), ffprobe n5.1.2-9-g807afa59cc-20221210, phantomjs 2.1.1
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.09.24, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1723 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.11.11, Current version: 2022.11.11
yt-dlp is up to date (2022.11.11)
[debug] [generic] Extracting URL: https://www.gmanetwork.com/fullepisodes/home/running_man_philippines/168677/running-man-philippines-catch-the-thief-full-chapter-2/video?section=home
[generic] video?section=home: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] video?section=home: Extracting information
[debug] Looking for embeds
ERROR: Unsupported URL: https://www.gmanetwork.com/fullepisodes/home/running_man_philippines/168677/running-man-philippines-catch-the-thief-full-chapter-2/video?section=home
Traceback (most recent call last):
  File "yt_dlp\YoutubeDL.py", line 1485, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1561, in __extract_info
  File "yt_dlp\extractor\common.py", line 674, in extract
  File "yt_dlp\extractor\generic.py", line 2721, in _real_extract
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.gmanetwork.com/fullepisodes/home/running_man_philippines/168677/running-man-philippines-catch-the-thief-full-chapter-2/video?section=home
```

### PR DESCRIPTION
**Pull request by HobbyistDev**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

This PR add site support for single video in gmanetwork.com. This PR assume all video either from Youtube or Dailymotion. 

Partially fixes #5770  (as this PR didn't include playlist extraction) 

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/gmanetwork.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+from .common import InfoExtractor
+from .dailymotion import DailymotionIE
+from .youtube import YoutubeIE
+
+
+class GMANetworkVideoIE(InfoExtractor):
+    _VALID_URL = r'https?://www\.gmanetwork\.com/\w+/\w+/\w+/(?P<id>\d+)/(?P<display_id>[\w-]+)/video'
+    _TESTS = [{
```
Comment (pukkandan) [inline yt_dlp/extractor/gmanetwork.py:8]: No test for dailymotion?

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/gmanetwork.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        if youtube_id:
+            return self.url_result(youtube_id, YoutubeIE)
+
+        # api call route
+        # more info at https://aphrodite.gmanetwork.com/fullepisodes/assets/fullepisodes/js/dist/fullepisodes_video.js?v=1.1.11
+        network_url = self._search_regex(
+            r'NETWORK_URL\s*=\s*[\'"](?P<url>[^\'"]+)', webpage, 'network_url')
+        json_data = self._download_json(f'{network_url}api/data/content/video/{content_id}', display_id)
+        if json_data.get('video_id'):
+            return self.url_result(json_data['video_file'], YoutubeIE)
```
[the remark this answers is not in the document]
Comment (HobbyistDev) [reply yt_dlp/extractor/gmanetwork.py:81]: Hmm, should we use video id from webpage or use youtube id ? Didn't it affect info-json id(s) ?

[the remark this answers is not in the document]
Comment (pukkandan) [reply yt_dlp/extractor/gmanetwork.py:81]: Youtube ID. It's just to get the id in --flat-playlist, and doesn't affect final result

File: yt_dlp/extractor/gmanetwork.py (hunk 1 above)
Comment (HobbyistDev) [reply yt_dlp/extractor/gmanetwork.py:8]: Is it a way to only restrict particular IE ? i used the `allowed_extractor` param in test but it always return ```yt_dlp.utils.DownloadError: ERROR: No suitable extractor (Youtube) found for URL 28BqW0AXPe0```. I don't have any links that only have `Dailymotion` url

Comment (pukkandan) [reply yt_dlp/extractor/gmanetwork.py:8]: Then how did you assume embeds are youtube/dailymotion?

Comment (HobbyistDev) [reply yt_dlp/extractor/gmanetwork.py:8]: It's from js source and manual testing as they have said `dailymotion_file`

File: yt_dlp/extractor/gmanetwork.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    }]
+
+    def _real_extract(self, url):
+        content_id, display_id = self._match_valid_url(url).group('id', 'display_id')
+        webpage = self._download_webpage(url, display_id)
+        # webpage route
+        youtube_id = self._search_regex(
+            r'var\s*YOUTUBE_VIDEO\s*=\s*[\'"]+(?P<yt_id>\w+)', webpage, 'youtube_id')
+        if youtube_id:
+            return self.url_result(youtube_id, YoutubeIE)
```
[the remark this answers is not in the document]
Comment (HobbyistDev) [reply yt_dlp/extractor/gmanetwork.py]: Fixed in 46d1c8d242950074a1a11da789847f8210f7f8f4

File: yt_dlp/extractor/gmanetwork.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
+from .common import InfoExtractor
+from .dailymotion import DailymotionIE
+from .youtube import YoutubeIE
+
+
+class GMANetworkVideoIE(InfoExtractor):
+    _VALID_URL = r'https?://www\.gmanetwork\.com/\w+/\w+/\w+/(?P<id>\d+)/(?P<display_id>[\w-]+)/video'
```
[the remark this answers is not in the document]
Comment (HobbyistDev) [reply yt_dlp/extractor/gmanetwork.py]: Fixed in 46d1c8d242950074a1a11da789847f8210f7f8f4

File: yt_dlp/extractor/gmanetwork.py (hunk 1 above)
Comment (pukkandan) [reply yt_dlp/extractor/gmanetwork.py:8]: > I don't have any links that only have Dailymotion url

Means you have a URL with both youtube and dailymotion? In that case, the extractor should return a playlist

Comment (HobbyistDev) [reply yt_dlp/extractor/gmanetwork.py:8]: Right, but didn't that would be considered unexpected by user as the link from Youtube and Dailymotion point to same video ?

Comment (pukkandan) [reply yt_dlp/extractor/gmanetwork.py:8]: Can you give an example?

Comment (HobbyistDev) [reply yt_dlp/extractor/gmanetwork.py:8]: Sure, for example this link (https://www.gmanetwork.com/fullepisodes/home/running_man_philippines/168677/running-man-philippines-catch-the-thief-full-chapter-2/video?section=home), from api request (`https://www.gmanetwork.com/api/data/content/video/168677`, the response return like this:
```json
{
  "video_id": "168677",
  "title": "Running Man Philippines: Catch the Thief (FULL CHAPTER 2)",
  "video_file": "28BqW0AXPe0",
  "video_image": "0",
  "description": "<p>Pa-intense na nang pa-intense ang laban sa 'Running Man Philippines!' Magawa kaya ng Runners na mabawi ang golden R-necklace mula kay Boobay? Mahuli na kaya nila ang spy sa grupo?<br />\n#RunningManPH #RunningManPhilippinesFullEpisode</p>\n\n<p>Consists of Full Episode 3 to 5:<br />\nRunning Man PH Full Episode 3 - aired Sept. 10, 2022<br />\nRunning Man PH Full Episode 4 - aired Sept. 11, 2022<br />\nRunning Man PH Full Episode 5 - aired Sept. 17, 2022</p>",
  "twitter_share": "0",
  "facebook_share": "0",
  "url": "tv/running_man_philippines/168677/running-man-philippines-catch-the-thief-full-chapter-2/video",
  "keywords": "youlol,running_man_philippines,glaiza_de_castro,ruru_madrid,buboy_villar,mikael_daez,lexi_gonzales,angel_guardian,kokoy_de_santos,full_episode,gma_network,gma_comedy,gma_entertainment,rss,rssfeed",
  "type": "1",
  "type_id": "1924",
  "post_date": "September 19, 2022",
  "auto_hide_date": "2022-09-19 17:53:42",
  "auto_hide_date_flag": "0",
  "comments": "0",
  "counter": null,
  "status": "1",
  "insert_timestamp": "2022-09-19 17:54:20",
  "update_timestamp": "2022-09-21 11:40:46",
  "count": null,
  "image": "FULL-EP-CMS_-20220919175418.jpg",
  "alt_text": "Running Man Philippines",
  "auto_url_flag": "0",
  "poll_id": "0",
  "poll_location": "0",
  "tile_expiry_date": "1950-11-30 00:00:00",
  "tile_type": "0",
  "display_in": "4",
  "pinned": "0",
  "pinned_expiry_date": "0000-00-00 00:00:00",
  "celebrity_news": "",
  "flipboard": "0",
  "is_ph": "0",
  "is_mobile": "1",
  "share_image": "FULL-EP-CMS_-20220919175418.jpg",
  "is_instant_article": "0",
  "author": "",
  "video_filename": "",
  "dailymotion": "0",
  "dm_selected": "0",
  "dailymotion_file": "k68aqqwX34H9SryiVOf",
  "geo_exception": "0",
  "default_source": "2",
  "image_orientation": "0",
  "background_position": "top center",
  "percentage": "100",
  "ga_author": "",
  "publish_date": "November 30, -0001 00:00:00",
  "fulleps_poster_image": "",
  "site_id": "1",
  "id": "168677",
  "contributors": [],
  "seo_title": "Running Man Philippines: Catch the Thief (FULL CHAPTER 2)",
  "socmed_title": "Running Man Philippines: Catch the Thief (FULL CHAPTER 2)",
  "page_title": "Running Man Philippines: Catch the Thief (FULL CHAPTER 2) | Running Man Philippines | TV",
  "override_url": "tv/running_man_philippines/168677/running-man-philippines-catch-the-thief-full-chapter-2/video",
  "kapuso_points": "0"
}
```
using yt-dlp for youtube id:
```shell
(python_proj2)> python -m yt_dlp -v "https://www.youtube.com/watch?v=28BqW0AXPe0"       
[debug] Command-line config: ['-v', 'https://www.youtube.com/watch?v=28BqW0AXPe0']
[debug] Portable config "yt-dlp\yt-dlp.conf": ['--netrc-location', 'yt-dlp_netrc']
[debug] User config: []
[debug] System config: []
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8 
[debug] yt-dlp version 2023.01.02 [d83b0ad80] (source) 
[debug] Lazy loading extractors is disabled
[debug] Git HEAD: 46d1c8d24 
[debug] Python 3.11.1 (CPython AMD64 64bit) - Windows-10-10.0.22000-SP0 (OpenSSL 1.1.1q  5 Jul 2022) 
[debug] exe versions: ffmpeg 5.1-full_build-www.gyan.dev (setts), ffprobe 5.1-full_build-www.gyan.dev, phantomjs 2.1.1 
[debug] Optional libraries: Cryptodome-3.14.1, certifi-2022.05.18.1, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3 
[debug] Proxy map: {} 
[debug] Loaded 1755 extractors 
[youtube] Extracting URL: https://www.youtube.com/watch?v=28BqW0AXPe0 
[youtube] 28BqW0AXPe0: Downloading webpage 
[youtube] 28BqW0AXPe0: Downloading android player API JSON 
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec:vp9.2, channels, acodec, lang, proto 
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), channels, acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
[debug] Default format spec: bestvideo*+bestaudio/best 
[info] 28BqW0AXPe0: Downloading 1 format(s): 248+251 
[debug] Invoking http downloader on "https://rr1---sn-uxa3vhnxa-uphl.googlevideo.com/videoplayback?expire=1672844737&ei=YUG1Y_7MBsKsigbS3om4Ag&ip=<ip>id=o-ACIslbe4te46CCqT-N1DseLzSFqu77G5jzX2_6g8tTz6&itag=248&source=youtube&requiressl=yes&mh=Nd&mm=31%2C29&mn=sn-uxa3vhnxa-uphl%2Csn-npoeenez&ms=au%2Crdu&mv=m&mvi=1&pcm2cms=yes&pl=24&initcwndbps=237500&spc=zIddbI0PRrYkx0jWvU_ChUEjdZwQnow&vprv=1&svpuc=1&mime=video%2Fwebm&gir=yes&clen=1337121086&dur=5312.500&lmt=1668869661128499&mt=1672822859&fvip=1&keepalive=yes&fexp=24007246&c=ANDROID&txp=5535434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AOq0QJ8wRQIhAPDIt6RiCdbsucTjzrXN59Dv_OR2JaTIdOIh7Ri2Mhv2AiA9KadGaQyiRLjLYEwHVCzcDzBrIetEYFsQZMBBBxxv0A%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpcm2cms%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRQIhAJI9wzYD5BQ6LqOIpohqYP4XmkB7ymG6Owp1DM1CtXmhAiB-nV8AnXVXHDKXPhKRky7IT-7F9Bcb8BeHULWH5nwTGA%3D%3D"
[debug] File locking is not supported. Proceeding without locking 
[download] Destination: Running Man Philippines： Catch the Thief (FULL CHAPTER 2) [28BqW0AXPe0].f248.webm 
[download]   0.9% of    1.25GiB at    2.41MiB/s ETA 08:44
ERROR: Interrupted by user
```

yt-dlp log for dailymotion:
```shell
$ yt-dlp -v "https://www.dailymotion.com/video/k68aqqwX34H9SryiVOf"
[debug] Command-line config: ['-v', 'https://www.dailymotion.com/video/k68aqqwX34H9SryiVOf']
[debug] User config: []
[debug] System config: []
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2023.01.02 [d83b0ad] (win32_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.22000-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg 5.1-full_build-www.gyan.dev (setts), ffprobe 5.1-full_build-www.gyan.dev, phantomjs 2.1.1
[debug] Optional libraries: Cryptodome-3.16.0, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1754 extractors
[dailymotion] Extracting URL: https://www.dailymotion.com/video/k68aqqwX34H9SryiVOf
[dailymotion] Downloading Access Token
[dailymotion] k68aqqwX34H9SryiVOf: Downloading media JSON metadata
[dailymotion] k68aqqwX34H9SryiVOf: Downloading metadata JSON
[dailymotion] k68aqqwX34H9SryiVOf: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] k68aqqwX34H9SryiVOf: Downloading 1 format(s): http-1080-1
[debug] Invoking http downloader on "https://proxy-33.sv4.dailymotion.com/sec(jZ16B8kLT6Yl5mIFpOtsdN6Zhuj8elbxrt-5oDhKHOnEY5DAciWlGMaA421Mlg5m9v_MKuDzVXTrmZWWLTqqug)/video/155/709/506907551_mp4_h264_aac_fhd.mp4"
[debug] File locking is not supported. Proceeding without locking
[download] Destination: Running Man Philippines： Catch the Thief (FULL CHAPTER 2) [k68aqqwX34H9SryiVOf].mp4
[download]   0.6% of    3.84GiB at    2.63MiB/s ETA 24:44
ERROR: Interrupted by user

```

File: yt_dlp/extractor/gmanetwork.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            'uploader_url': 'http://www.youtube.com/user/GMANETWORK',
+        }
+    }]
+
+    def _real_extract(self, url):
+        content_id, display_id = self._match_valid_url(url).group('id', 'display_id')
+        webpage = self._download_webpage(url, display_id)
+        # webpage route
+        youtube_id = self._search_regex(
+            r'var\s*YOUTUBE_VIDEO\s*=\s*[\'"]+(?P<yt_id>\w+)', webpage, 'youtube_id')
```
Comment (bashonly) [inline yt_dlp/extractor/gmanetwork.py:71]: this should be non-fatal if we want to be able to use the API call route

Comment (bashonly) [inline yt_dlp/extractor/gmanetwork.py:71]: regex fix + this should be non-fatal if we want to be able to use the API call route
```suggestion
        youtube_id = self._search_regex(
            r'var\s*YOUTUBE_VIDEO\s*=\s*[\'"]+(?P<yt_id>[\w-]+)', webpage, 'youtube_id', fatal=False)
```

**Pull request merged**